Add more rows to some tables

ediosyncratic opened the pull request:
Penny has too many weapons for the current four slots, so I've been resorting to documenting weapon properties and details in possessions rows, of which there are barely enough too.  So let's have some more rows.

ediosyncratic committed:
Extend "Other Possessions" to 37 rows

It (along with Feats & Special Abilities) sits alongside the Skills
table, which is tall enough to leave space for this.

ediosyncratic committed:
Enable more rows in the Weapon and Armor tables

A character with diverse different ways to attack can easily have more
than four weapons. Since the table to enable new rows aligns Armor
with Weapons, include as meany for Protective Items as for Weapons,
even though that may be less relevant. Allow the possibility of a
secondary shield as one of the new Armor slots.

This naturally widens the space these enablers take up, so widen their
cells of the table in which they appear, separating new class enablers
from their old class wide. The other two class=wide columns in the
table have ample room (their content is a number unlikely to make it
to three digits, much less more; so their titles are the constraints
on width), so we can pilfer 5% of total width off each of those and
give it to the enablers' column.

In the process, move the styling of the check-boxes from a style
attribute on each of them to a suitable CSS rule; and add one more
&nbsp; to the Armor label to (hopefully) fix the alignment of the
checkboxes in the two rows.  (I, at least, see the Armor row's
check-boxes visible to the left of the Weapon row's.)  Also clear out
some trash (blank lines, stray comment) and make some doc comment a
bit more truthful / coherent.

Since enablers share their column with the spell resistance entry, we
can now give that its full name instead of truncating to SPELL RESIST.

ediosyncratic committed:
Use loops for AC calculations

... rather than having four copy-and-paste blocks doing the same thing
(except with a quirk for 2, which is the shield slot) with different
digits in the middle of an element name. This will make life simpler
when I change how many of them there are ...

ediosyncratic committed:
Fix typo in name of weapon-crit cell name

When someone converted the code from dealing with each of Weapon1Crit,
Weapon2Crit and Weapon3Crit independently to using "Weaopon" + i +
"1Crit" they rather obviously failed to delete the 1 from the start of
that last. Do so now, but include a kludge-around so that those with
data saved with the typo do get it looked up with the typo'd name (but
eventually stored with the right name).

ediosyncratic commented on public_profiler/v3.5-DD/ac.js:
```diff
@@ -67,32 +67,23 @@ function ACChangeArmor()
   {
       sheet().Armor2Carried.checked = true;
-     sheet().ACShield.value = GetNum(sheet().Armor2Bonus);
+      sheet().ACShield.value = GetNum(sheet().Armor2Bonus);
```
Oh, look - the above is wrongly covering i = 2; it needs an adapted version to handle that case differently.

ediosyncratic commented on root_profiler/engine/sheet_globals.php:
```diff
@@ -7,18 +7,24 @@
-    echo "name=\"$name\" value=\"" . $DATA[$name] . "\"";
-    echo ' title="' . $DATA[$name] . '"';
+    $value = kludgedLookupInData($name);
+    echo 'name="$name" value="$value" title="$value"';
```
ah - oops -  PHP is like Unix shells, not Python - single-quoted strings suppress variable expansion.

julien-picalausa commented on public_profiler/v3.5-DD/ac.js:
```diff
@@ -67,32 +67,20 @@ function ACChangeArmor()
+  for (var i = 1; i <= 8; i++) {
+      if (sheet()["Armor" + i + "Worn"].checked) {
+          sheet()["Armor" + i + "Carried"].checked = true;
+          if (i == 2)
```
With 8 slots, it might be useful to have two slots that count as shield, on the theoretical possibility that someone might have two shields and swap between them.

ediosyncratic replied:
Fair enough - I even considered the possibility of adding a second but passed thinking "why, though ?" - which you've answered.  I guess i % 4 == 2 will do as test ...

ediosyncratic committed:
Clean up v3.5-DD.php: spacing and case

Fix various upper-case HTML tags to be lower-case.
Fix various pieces of inconsistent indentation and spacing.

ediosyncratic committed:
Make coding style less clunky in a few places

Save repeated evaluation of the same expression, ditch braces on
single-line bodies, use single-quoted strings to avoid needing to
escape inner double quotes (where no variable expansion was needed
within).

ediosyncratic committed:
Make notes a class instead of a duplicated id

HTML and XML don't allow two elements with the same id.
The "Private Notes" shared an id with the public "Other Notes".

ediosyncratic committed:
Tag table-end (and some div-end) elements with where they started

This makes it easier to find which thing ends where, given that the
indentation is a bit haphazard.

ediosyncratic committed:
Fix broken syntax in ogl/skills.js

I seem to have left a stray word in the midst of the sillsHelpURL
data, back in commit b2489899e7a0076014cd9c1052d0638c955e8f93

ediosyncratic committed:
Only show the first two armor and weapons on a new sheet.

ediosyncratic committed:
Include off-person containers in initial display.

ediosyncratic commented:
Minor note to help make sense of the off-person containers change.

julien-picalausa commented on public_profiler/v3.5-DD/ac.js:
```diff
@@ -59,40 +59,29 @@ function ACCalc()
-      sheet().Armor2Carried.checked = true;
-     sheet().ACShield.value = GetNum(sheet().Armor2Bonus);
+
+  for (var i = 1; i <= 8; i++) {
```
Might be a good idea to have a global constant for this, instead of just using 8 throughout.
